homeassistant/components/fibaro/binary_sensor.py

```diff
@@ -66,4 +73,38 @@ def __init__(self, fibaro_device: Any) -> None:
 
     def update(self) -> None:
         """Get the latest data and update the state."""
-        self._attr_is_on = self.current_binary_state
+        if self.device_class == BinarySensorDeviceClass.MOVING:
+            self._attr_is_on = self._is_moving()
```

Comment:
I'd call `_get_moving_values` once here and pass the values to helpers that can return values for `_attr_is_on` and `_attr_extra_attributes` and return the latter from the `extra_state_attributes` property.

Reply:
My opinion is that this does not make much sense as the base fibaro entity also provide extra state attributes. Because of that it is not a good idea to set `_attr_extra_state_attributes` as it gives some strange combinations of using the method and the attribute. I checked several other integrations and all which have a base entity which as well provide some extra state attributes use method overwriting so that it is clear that the base entity as well provides some attributes.

Reply:
We don't have to use `_attr_extra_state_attributes` but I think we should try to avoid deserializing twice if possible. We can use another instance attribute that we update in `update`.

Reply:
Makes sense, I changed it. Thanks fro your help.